crypto/rand/randfile.c

```diff
@@ -262,11 +262,9 @@ const char *RAND_file_name(char *buf, size_t size)
         }
     }
 #else
-    if (OPENSSL_issetugid() != 0) {
+    if ((s = ossl_safe_getenv("RANDFILE")) == NULL || *s == '\0') {
```

Comment:
Problem here is that suggested ossl_safe_getenv is "tri-state", denied, no-variable, value, and it's impossible to tell first two apart. As result of suggested change no-RANDFILE-variable is rendered to be equivalent of "denied". No good...

Reply:
I don't see what's the problem here.

Reply:
Yeah, "no good" was too strong. Formally speaking the flow *is* different because of "tri-state" thing, but it's true that overall outcome is the same in this case. Modulo redundant call to set[gu]id in case ossl_safe_getenv('RANDFILE") returns NULL...